File: 剑指offer代码自编/49_strToInt.py
def strToInt(a):
	if len(a) == 0 or a == None:
		return 
	num = 0
	if a[0] == '+'：
		positive = True
	if a[0] == '-':
		positive = False
	if a[1]:
		num = strToIntCore(a[1:], positive)
	return num	

def strToIntCore(a, positive):
	sum = 0
	i = 0
	while(a[i]):
		if '0'<= a[i] <= '9':
			if positive:
				flag = 1
			else:
				flag = -1
			sum = sum * 10 + flag * (a[i] - '0')
			i += 1
		else:
			sum = 0
			break
	return sum;

File: 剑指offer代码/49_strToInt.cpp
int strToInt(char* a)
{
	if(a != NULL and *a != '\0')
	{
		bool positive = true;
		int num = 0;
		if(*a == '+')
			a++;
		if(*a == '-')
			positive = false;
			a++;
		if(*a != '\0')
			num = strToIntCore(a, positive);
	}
	return sum;
}

int strToIntCore(char* a, bool positive)
{
	int i = 0;
	int num = 0;
	while(*a != '\0')
	{
		if(*a >= '0' and *a <= '9')
		{
			int flag = positive ? 1 : -1;
			num = num * 10 + flag * (*a - '0');
			a++;
		}
		else
		{
			num = 0;
			break;
		}
	}
	return num;
}